### ISSUE TITLE
Accessibility testing of Accordion header fix

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#12352]
## Description
Ticket represents asking LauraF to test plugin update created in #12342 in order to satisfy & resolve the issue originally reported in #11205.

This is a manual "smoke test" in Staging.

## Acceptance criteria

- [x] Manual testing in review instance of header level violations confirms that the plugin is fixing existing violations
- [x] Manual testing in review instance to confirm plugin change is _not_ modifying headers that it shouldn't or creating new violations
- [x] Automated testing in staging/prod once merged to compare tests after plugin

**Comment by laflannery** [comment]
@jtmst I have more questions as I have been looking at which pages/accordions to test.  Like we discussed, these scenarios don't necessarily need to be covered with the new plugin I just want to make sure I am clear on what to test.

**Will the below scenarios be considered with the new plugin?**

- The `<div data-template="paragraphs/wysiwyg">` is not the immediate prior sibling of the `<div data-template="paragraphs/collapsible_panel">` but instead it is contained within a "Feature" `<div class="feature">`
  - Example: https://www.va.gov/richmond-health-care/work-with-us/internships-and-fellowships/occupational-therapy-fellowship-program/
- The immediate prior `<div data-template="paragraphs/wysiwyg">` sibling does not contain any header elements that the plugin can use to determine the proper heading level for the accordions
   - Example: https://www.va.gov/cincinnati-health-care/programs/homeless-veterans-program/
- The immediate prior `<div data-template="paragraphs/wysiwyg">` sibling is completely blank
   - Example: https://www.va.gov/maryland-health-care/work-with-us/internships-and-fellowships/pharmacy-residency-programs/
- Prior to the `<div data-template="paragraphs/collapsible_panel">` there is no `<div data-template="paragraphs/wysiwyg">` at all (My theory is that this scenario will not apply or be impacted by the plugin at this time)
  - Example: https://www.va.gov/cincinnati-health-care/programs/beneficiary-travel-self-service-system-btsss/ 
- The `<div data-template="paragraphs/wysiwyg">` is not the immediate prior sibling of the `<div data-template="paragraphs/collapsible_panel">` but it is interrupted by another type of div
  - Example: https://www.va.gov/augusta-health-care/work-with-us/internships-and-fellowships/psychology-training-programs/

A couple examples of the above scenarios being combined:
- Example of `<div data-template="paragraphs/wysiwyg">`contained within a Feature div but also has no heading: https://www.va.gov/maryland-health-care/work-with-us/internships-and-fellowships/pharmacy-residency-programs/
- Example of `<div data-template="paragraphs/wysiwyg">`contained within a Feature div, also has no heading and also not the immediate prior sibling: https://www.va.gov/columbia-south-carolina-health-care/programs/my-healthevet/ (There's a lot happening here, my guess is this wouldn't not be covered in the plugin)

**Comment by wesrowe** [comment]
@jtmst @laflannery – do I understand correctly that this is to be done in Prod? I.e., Laura will smoke-test manually in Staging before merge...

If so, I'll make #12346 a blocker for this ticket. And I wonder, do we need a ticket for "Laura smoke test in Staging"?

**Comment by laflannery** [comment]
@wesrowe I will do my manual testing in PR, on the review instance, as that is the environment that my scanning tools cannot reach. They can reach staging and Prod - which means I can do much more comprehensive automated, testing once we move everything to staging.

You are correct in that we probably need 2 tickets - however because of the AC on this one says "Spot Check" I think I was thinking this was the manually testing ticket? Let me know if that helps clarify the process and if if helps you frame how you want to frame the tickets.

**Comment by wesrowe** [comment]
@apisandipas, the question above is for you

**Comment by apisandipas** [comment]
@laflannery  Answers below. With some additional questions as well. 

- The <div data-template="paragraphs/wysiwyg"> is not the immediate prior sibling of the <div data-template="paragraphs/collapsible_panel"> but instead it is contained within a "Feature" <div class="feature">
Example: https://www.va.gov/richmond-health-care/work-with-us/internships-and-fellowships/occupational-therapy-fellowship-program/

**Answer: This scenario has been accounted for in the plugin.**

- The immediate prior <div data-template="paragraphs/wysiwyg"> sibling does not contain any header elements that the plugin can use to determine the proper heading level for the accordions
Example: https://www.va.gov/cincinnati-health-care/programs/homeless-veterans-program/

**Answer: This scenario has been accounted for in the plugin.**


- The immediate prior <div data-template="paragraphs/wysiwyg"> sibling is completely blank
Example: https://www.va.gov/maryland-health-care/work-with-us/internships-and-fellowships/pharmacy-residency-programs/

**Question**: What is the desired heading level for the Accordions in this case? Should they also be H3's? I'm not certain it would be possible to set up in an automated way if they dont have an associated header in this instance, it attempts to associate them based on the previous header, which is an h3 itself.

- Prior to the <div data-template="paragraphs/collapsible_panel"> there is no <div data-template="paragraphs/wysiwyg"> at all (My theory is that this scenario will not apply or be impacted by the plugin at this time)
Example: https://www.va.gov/cincinnati-health-care/programs/beneficiary-travel-self-service-system-btsss/

**Answer: This scenario has been accounted for in the plugin.**

- The <div data-template="paragraphs/wysiwyg"> is not the immediate prior sibling of the <div data-template="paragraphs/collapsible_panel"> but it is interrupted by another type of div
Example: https://www.va.gov/augusta-health-care/work-with-us/internships-and-fellowships/psychology-training-programs/

**Answer: This scenario has been accounted for in the plugin.**

**Comment by jilladams** [comment]
We added this to S78 during planning, but Laura flagged that it's blocked til the plugin bug is fixed, so: moving back to stretch til unblocked, with intent to pull it in asap in sprint.

**Comment by wesrowe** [comment]
>I need to ask about this:

@laflannery, are you asking? 🤣

**Comment by laflannery** [comment]
@wesrowe It was a random thought I had and I didn't want to forget! It was really just a note for myself, maybe I should have been clearer in that hahaha But I asked @jtmst and he and I are both keeping this in mind for the next round of updates/testing

**Comment by laflannery** [comment]
### Prior to plugin pa11y-ci tests
**Notes:**

- These tests are looking for ALL heading violations, which is one of the reasons we needed to rely so heavily on manual testing for this. We are not looking for scans with zero errors, because there are so many editor created issues that remain.  The goal is to make sure we don't have a significant increase in errors in the scanning before the plugin and after.
- Scans are from Prod VA.gov (I can run staging tomorrow if needed)

**Report of URLs that I had used for the manual testing:**
![image](https://user-images.githubusercontent.com/106678594/222286045-9e7ba82c-db15-4918-91e3-3a7720cff4d9.png)

**Reports of random sets of 1000 URLs:**
It takes about 1 hours to run a set of 1000 URLs which is why I broke it up this way.
- (will add these tomorrow)

**Comment by laflannery** [comment]
Run all the above tests in staging and put results here:
